test/common/websocket/codec_test.cc

```diff
@@ -0,0 +1,824 @@
+#include <array>
+#include <cstdint>
+#include <string>
+#include <vector>
+
+#include "source/common/buffer/buffer_impl.h"
+#include "source/common/websocket/codec.h"
+
+#include "test/common/buffer/utility.h"
+#include "test/test_common/printers.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace WebSocket {
+namespace {
+
+// A single-frame unmasked text message
+// 0x81 0x05 0x48 0x65 0x6c 0x6c 0x6f (contains "Hello")
+//                   H        e        l        l        o
+// 10000001 00000101 01001000 01100101 01101100 01101100 01101111
+// fin - 1
+// opcode - 0x1 - text data
+// mask bit - 0
+// length - 5 bytes 0000101
+TEST(WebSocketCodecTest, decodeSingleUnmaskedFrame) {
+  Buffer::OwnedImpl buffer;
+  Buffer::addSeq(buffer, {0x81, 0x05, 0x48, 0x65, 0x6c, 0x6c, 0x6f});
+
+  Decoder decoder;
+  absl::optional<std::vector<Frame>> frames = decoder.decode(buffer);
+
+  ASSERT_TRUE(frames.has_value());
+  EXPECT_EQ(false, frames.value()[0].masking_key_.has_value());
+  EXPECT_EQ(1, frames.value().size());
+  EXPECT_EQ(5, frames.value()[0].payload_length_);
```

Comment:
This chunk of EXPECT_EQs does not appear to be checking that all the fields are correct. In particular, it's missing `final_fragment_` and `opcode_`. Since we're going to do lots of checking of frames, I wonder if it might make sense do add a function like:
```
bool areFramesEqual(const Frame& a, const Frame& b) {
  return a.final_fragment_ == b.final_fragment_ &&
    a.opcode_ == b.opcode_ &&
    a.masking_key_ == b.masking_key_ &&
    a.payload_length_ == b.payload_length_ &&
    a.payload_.toString() == b.payload.toString()
}
```
Then we can use it here like:
```
EXPECT_TRUE(areFramesEqual(*frames[0], {false, kFrameOpcodeText, {}, 5, "Hello"}));
```
Or something like that.